onnx_xla/onnx_xla_interface.cc

```diff
@@ -1,26 +1,42 @@
 #include "onnx/onnxifi.h"
 #include "onnx_xla/backend.h"
+#include <thread>
+#include <mutex>
+#include <condition_variable>
 
 //TODO: Robust error handling
-//TODO: Implement Event functions
 //TODO: Figure out how to determine type of device, what information to store
 //      about hardware, and how to modify execution as a result
 
-#define ONNXIFI_CATCH_EXCPETION()                                              \
+#define ONNXIFI_TRY_CATCH(TRY_BLOCK)                                           \
```

Comment:
If we are going to do this, we can just make `ONNXIFI_TRY_CATCH()` a function.

Reply:
made into function accepting std::function<onnxStatus()> as argument